Framework/DataObjects/inc/MantidDataObjects/CoordTransformDistance.h

```diff
@@ -38,29 +38,36 @@ DECLARE_VECTOR_PARAMETER(DimensionsUsedVectorParam, bool)
  */
 class DLLExport CoordTransformDistance : public Mantid::API::CoordTransform {
 public:
-  CoordTransformDistance(const size_t inD, const coord_t *center,
-                         const bool *dimensionsUsed, const size_t outD = 1);
+  CoordTransformDistance(
+      const size_t inD, const coord_t *center, const bool *dimensionsUsed,
+      const size_t outD = 1,
+      const std::vector<Kernel::V3D> directions = std::vector<Kernel::V3D>(0),
+      const std::vector<double> abcRadii = std::vector<double>(0, 0.0));
+
   CoordTransform *clone() const override;
-  ~CoordTransformDistance() override;
   std::string toXMLString() const override;
   std::string id() const override;
 
   void apply(const coord_t *inputVector, coord_t *outVector) const override;
 
   /// Return the center coordinate array
-  const coord_t *getCenter() { return m_center; }
+  const std::vector<coord_t> getCenter() { return m_center; }
```

Comment:
We've made this a bit more expensive now as previously it just returned a pointer. Can we do `const std::vector<coord_t> & getCenter()` instead so it returns without copy?
Similar for the `getDimensionsUsed`?